### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by zhixingheyi-tian**
## What changes were proposed in this pull request?

Update readme  for OAP-0.5

## How was this patch tested?

N/A

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,7 +46,7 @@ For a more detailed examples with performance compare, you can refer to [this pa
 
 To run all the tests, use
 ```
-mvn test
+mvn clean -q -Pspark-2.3 -Ppersistent-memory test
```
Comment (LuciferYang) [inline README.md]: run test must need `persistent-memory` profile ?

Comment (zhixingheyi-tian) [reply README.md]: Unit tests contain the compile.

thanks

File: README.md [hunk 2]
```diff
@@ -3,27 +3,28 @@
 
 OAP - Optimized Analytics Package (previously known as Spinach) is designed to accelerate Ad-hoc query. OAP defines a new parquet-like columnar storage data format and offering a fine-grained hierarchical cache mechanism in the unit of “Fiber” in memory. What’s more, OAP has extended the Spark SQL DDL to allow user to define the customized indices based on relation.
 ## Building
-By defaut, it builds for Spark 2.1.0. To specify the Spark version, please use profile spark-2.1 or spark-2.2.
+By defaut, it builds for Spark 2.1.0. To specify the Spark version, please use profile spark-2.1 , spark-2.2 or spark-2.3.
```
Comment (adrian-wang) [inline README.md]: spark-2.1, spark-2.2 or spark-2.3.

File: README.md [hunk 3]
```diff
@@ -3,27 +3,28 @@
...
 ```
-mvn -DskipTests package
-mvn -DskipTests -Pspark-2.2 package
+mvn clean -q -Ppersistent-memory -DskipTests package
+mvn clean -q -Pspark-2.2 -Ppersistent-memory -DskipTests package
+mvn clean -q -Pspark-2.3 -Ppersistent-memory -DskipTests package
 ```
 ## Prerequisites
-You should have [Apache Spark](http://spark.apache.org/) of version 2.1.0 or 2.2.0 installed in your cluster. Refer to Apache Spark's [documents](http://spark.apache.org/docs/2.1.0/) for details.
+You should have [Apache Spark](http://spark.apache.org/) of version 2.1.0, 2.2.0 or 2.3.0 installed in your cluster. Refer to Apache Spark's [documents](http://spark.apache.org/docs/2.1.0/) for details.
```
Comment (adrian-wang) [inline README.md]: do we need 2.2.0, or 2.2.x? also the reference link is specfic to 2.1.0

File: README.md [hunk 4]
```diff
@@ -94,7 +99,7 @@ Row Group Size - Row count for each row group
 
 Compression Codec - Choose compression type for OAP data files.
 * Default: GZIP
-* Values: UNCOMPRESSED, SNAPPY, GZIP, LZO
+* Values: UNCOMPRESSED, SNAPPY, GZIP, LZO(Note that ORC does not support GZIP)
```
Comment (adrian-wang) [inline README.md]: whitespace before `(`

File: README.md [hunk 5]
```diff
@@ -3,27 +3,28 @@
...
+mvn clean -q -Pspark-2.2 -Ppersistent-memory -DskipTests package
+mvn clean -q -Pspark-2.3 -Ppersistent-memory -DskipTests package
 ```
 ## Prerequisites
-You should have [Apache Spark](http://spark.apache.org/) of version 2.1.0 or 2.2.0 installed in your cluster. Refer to Apache Spark's [documents](http://spark.apache.org/docs/2.1.0/) for details.
+You should have [Apache Spark](http://spark.apache.org/) of version 2.1.0, 2.2.0 or 2.3.0 installed in your cluster. Refer to Apache Spark's [documents](http://spark.apache.org/docs/2.1.0/) for details.
 ## Use OAP with Spark
-1. Build OAP, `mvn -DskipTests package` and find `oap-<version>.jar` in `target/`
-2. Deploy `oap-<version>.jar` to master machine.
+1. Build OAP find `oap-<version>-with-<spark-version>.jar` in `target/`
```
Comment (adrian-wang) [inline README.md]: Build OAP, find ...

File: README.md (hunk 2 above)
Comment (LuciferYang) [reply README.md]: Welcome Teacher Wang back~

File: README.md (hunk 4 above)
Comment (zhixingheyi-tian) [reply README.md]: Good catch,
Thanks

**Pull request merged**